src/garage/tf/models/categorical_gru_model.py

```diff
@@ -0,0 +1,113 @@
+            tf.Tensor.
+        hidden_state_init (callable): Initializer function for the
+            initial hidden state. The functino should return a tf.Tensor.
+        hidden_state_init_trainable (bool): Bool for whether the initial
+            hidden state is trainable.
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    """
+
+    def __init__(self,
+                 output_dim,
+                 hidden_dim,
+                 name=None,
+                 hidden_nonlinearity=tf.nn.tanh,
+                 hidden_w_init=tf.glorot_uniform_initializer(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 recurrent_nonlinearity=tf.nn.sigmoid,
+                 recurrent_w_init=tf.glorot_uniform_initializer(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.glorot_uniform_initializer(),
+                 output_b_init=tf.zeros_initializer(),
+                 hidden_state_init=tf.zeros_initializer(),
+                 hidden_state_init_trainable=False,
+                 layer_normalization=False):
+        super().__init__(
+            output_dim=output_dim,
+            hidden_dim=hidden_dim,
+            name=name,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            recurrent_nonlinearity=recurrent_nonlinearity,
+            recurrent_w_init=recurrent_w_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            hidden_state_init=hidden_state_init,
+            hidden_state_init_trainable=hidden_state_init_trainable,
+            layer_normalization=layer_normalization)
+
+    def network_output_spec(self):
+        """Network output spec.
+
+        Returns:
+            list[str]: Name of the model outputs, in order.
+
+        """
+        return [
+            'all_output', 'step_output', 'step_hidden', 'init_hidden', 'dist'
+        ]
+
+    # pylint: disable=arguments-differ
+    def _build(self, state_input, step_input, step_hidden, name=None):
+        """Build model.
+
+        Args:
+            state_input (tf.Tensor): Full observation input.
+            step_input (tf.Tensor): Step observation input.
+            step_hidden (tf.Tensor): Hidden state for step.
+            name (str): Name of the model, also the name scope.
+
+        Returns:
+            tf.Tensor: Entire time-series outputs.
+            tf.Tensor: Step output.
+            tf.Tensor: Step hidden state.
+            tf.Tensor: Initial hidden state.
+            tfp.distributions.OneHotCategorical: Distribution.
+
+        """
+        prob, step_output, step_hidden, init_hidden = super()._build(
+            state_input, step_input, step_hidden, name=name)
+        dist = tfp.distributions.OneHotCategorical(prob)
+        return prob, step_output, step_hidden, init_hidden, dist
```

Comment:
is there a reason to include prob when i can just do `dist.logits` instead?

Reply:
fixed